File: frontend/src/services/monitoring.ts
import api from './api'

export interface MonitoringOverview {
  total_projects: number
  active_projects: number
  risk_projects: number
  pending_gates: number
  total_executions: number
}

export interface ProjectStats {
  stage_count: number
  execution_count: number
  gate_count: number
  log_count: number
}

export interface OperationLog {
  log_id: string
  project_id: string
  operator_id: string | null
  action: string
  target_type: string | null
  target_id: string | null
  detail: string | null
  created_at: string | null
}

export interface OperationLogList {
  logs: OperationLog[]
  total: number
}

export async function fetchMonitoringOverview(): Promise<MonitoringOverview> {
  const res = await api.get<MonitoringOverview>('/v1/monitoring/overview')
  return res.data
}

export async function fetchProjectStats(projectId: string): Promise<ProjectStats> {
  const res = await api.get<ProjectStats>(`/v1/monitoring/projects/${projectId}/stats`)
  return res.data
}

export async function fetchOperationLogs(projectId: string, limit = 50): Promise<OperationLog[]> {
  const res = await api.get<OperationLogList>(`/v1/monitoring/projects/${projectId}/operation-logs`, {
    params: { limit },
  })
  return res.data.logs
}

File: backend/app/schemas/monitoring.py
"""Pydantic DTOs for monitoring operations."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class MonitoringOverviewDTO(BaseModel):
    """Global monitoring overview response."""

    total_projects: int
    active_projects: int
    risk_projects: int
    pending_gates: int
    total_executions: int


class ProjectStatsDTO(BaseModel):
    """Per-project monitoring stats response."""

    stage_count: int
    execution_count: int
    gate_count: int
    log_count: int


class OperationLogDTO(BaseModel):
    """Operation log entry response."""

    log_id: str
    project_id: str
    operator_id: str | None
    action: str
    target_type: str | None
    target_id: str | None
    detail: str | None
    created_at: datetime | None


class OperationLogListDTO(BaseModel):
    """Paginated operation log list response."""

    logs: list[OperationLogDTO]
    total: int
